lldb/source/Core/Mangled.cpp

```diff
@@ -211,7 +211,7 @@ void Mangled::SetValue(ConstString s, bool mangled) {
 
 void Mangled::SetValue(ConstString name) {
   if (name) {
-    if (cstring_is_mangled(name.GetStringRef())) {
+    if (cstring_is_mangled(name.AsCString())) {
```

Comment:
This may look like a regression at first glance, but this avoids an extra copy of the string, since we need to pass it to a Swift API that takes a `const char *`.

Reply:
I'm not sure I follow. A few questions:
 - Should SwiftLanguageRuntime::IsSwiftMangledName() take a StringRef?
 - The comment in cstring_is_mangled implies that ConstString::GetStringRef() doesn't always return a null-terminated string. Is this correct?

Reply:
`SwiftLanguageRuntime::IsSwiftMangledName()` calls into a Swift API that takes a `const char *`; so changing that to a StringRef would incur an extra copy to ensure the NUL-terminator. ConstString are; to my knowledge always NUL-terminated.

Reply:
So I don't understand what calling `AsCString()` instead of `GetStringRef()` fixes. Given `cstring_is_mangled()` takes a `StringRef`, what this achieves is creating a temporary one which will need to `strlen` the passed `char*`. What am I missing?

Reply:
Gotcha. I was confused.